integrationtests/controller/bundle/status_test.go

```diff
@@ -0,0 +1,176 @@
+				if err != nil {
+					return err
+				}
+				bd.Status.Display.State = "Ready"
+				bd.Status.AppliedDeploymentID = bd.Spec.DeploymentID
+				bd.Status.Ready = true
+				bd.Spec.StagedDeploymentID = bd.Spec.DeploymentID
+				bd.Status.NonModified = true
+				return k8sClient.Status().Update(ctx, bd)
+			}).ShouldNot(HaveOccurred())
+
+			err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bd)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bd.Status.Display.State).To(Equal("Ready"))
+			Expect(bd.Status.AppliedDeploymentID).To(Equal(bd.Spec.DeploymentID))
+
+			Eventually(func() bool {
+				err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bundle)
+				Expect(err).NotTo(HaveOccurred())
+				return bundle.Status.Summary.Ready == 1
+			}).Should(BeTrue())
+		})
+	})
+
+	When("Cluster changes", func() {
+		BeforeEach(func() {
+			cluster, err := createCluster("cluster", namespace, nil, namespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cluster).To(Not(BeNil()))
+			targets := []v1alpha1.BundleTarget{
+				{
+					BundleDeploymentOptions: v1alpha1.BundleDeploymentOptions{
+						TargetNamespace: "targetNs",
+					},
+					Name:        "cluster",
+					ClusterName: "cluster",
+				},
+			}
+			bundle, err := createBundle("name", namespace, targets, targets)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bundle).To(Not(BeNil()))
+		})
+
+		AfterEach(func() {
+			Expect(k8sClient.Delete(ctx, &v1alpha1.Bundle{ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: namespace,
+			}})).NotTo(HaveOccurred())
+
+		})
+
+		It("updates the status fields", func() {
+			cluster := &v1alpha1.Cluster{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "cluster"}, cluster)
+			Expect(err).NotTo(HaveOccurred())
+			resourceVersion := cluster.ResourceVersion
+
+			bundle := &v1alpha1.Bundle{}
+			Eventually(func() bool {
+				err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bundle)
+				Expect(err).NotTo(HaveOccurred())
+				return bundle.Status.Summary.Ready == 0
+			}).Should(BeTrue())
+
+			// prepare bundle deployment so it satisfies the status change
+			bd := &v1alpha1.BundleDeployment{}
+			Eventually(func() error {
+				err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bd)
+				if err != nil {
+					return err
+				}
+				return nil
+			}).ShouldNot(HaveOccurred())
+
+			By("To reflect the 'Ready' status in bundle, bundleDeployment needs below mentioned status fields.")
+			Eventually(func() error {
+				err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bd)
+				bd.Status.Display.State = "Ready"
+				bd.Status.AppliedDeploymentID = bd.Spec.DeploymentID
+				bd.Status.Ready = true
+				bd.Spec.StagedDeploymentID = bd.Spec.DeploymentID
+				bd.Status.NonModified = true
+				return k8sClient.Status().Update(ctx, bd)
```

Comment:
The `Eventually` block is needed here because `update` is not always successful on the first try. The bundle deployment exists but the update fails with the following error ```the object has been modified; please apply your changes to the latest version```. That's why this update is in eventually block